src/integration_tests/test_with_instance.py

```diff
@@ -0,0 +1,128 @@
+import os, json
+from fabric.api import settings
+from tests import base
+from buildercore import bootstrap, cfngen, lifecycle, utils, core
+from buildercore.config import BOOTSTRAP_USER
+import cfn
+import logging
+
+logging.disable(logging.NOTSET) # re-enables logging during integration testing
+
+# the intent here is:
+# * launch an ec2 instance without running highstate
+# * run any logic against it that needs a running ec2 instance
+# * optionally leave the instance running while debugging happens
+
+class One(base.BaseCase):
+    @classmethod
+    def setUpClass(self): # cls, not self
+        super(One, self).setUpClass()
+
+        # to re-use an existing stack, ensure self.reuse_existing_stack is True
+        # this will read the instance name from a temporary file (if it exists) and
+        # look for that, creating it if doesn't exist yet
+        # also ensure self.cleanup is False so the instance isn't destroyed after tests complete
+        self.reuse_existing_stack = False
+        self.cleanup = True
+
+        self.stacknames = []
+        self.environment = base.generate_environment_name()
```

Comment:
this is key to allow isolation of the test suite and having multiple builds running simultaneously: each `stackname` has a random suffix and different test suites don't see each other

Reply:
I get that - I don't think this undermines that. `generate_environment_name` had to be shifted out of the class is all.

Reply:
Was a positive comment for anyone reading here in the future

Reply:
ah - sorry, I was a bit unsure. I'm also feeling slightly defensive as I prepare to take responsibility for builder breaking again so soon after the salt upgrade. let's hope it doesn't break ;) the last of the bugs are being ironed out and the tests are doing a good job of unearthing things. I didn't realize just how deeply boto2-isms had infiltrated the code.

Reply:
I have a feeling of why <img src="https://screenshotscdn.firefoxusercontent.com/images/2a26ad7f-4f5a-47ee-ae12-1422190ff3a3.png /> :grinning:

Reply:
heh.